bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisBoxHandler.java

```diff
@@ -0,0 +1,395 @@
+import org.openhab.binding.intesis.internal.api.IntesisBoxMessage;
+import org.openhab.binding.intesis.internal.api.IntesisBoxSocketApi;
+import org.openhab.binding.intesis.internal.config.IntesisBoxConfiguration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link IntesisBoxHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Cody Cutrer - Initial contribution
+ * @author Rocky Amatulli - additions to include id message handling, dynamic channel options based on limits.
+ * @author Hans-Jörg Merk - refactored for openHAB 3.0 compatibility
+ *
+ */
+@NonNullByDefault
+public class IntesisBoxHandler extends BaseThingHandler implements IntesisBoxChangeListener {
+
+    private final Logger logger = LoggerFactory.getLogger(IntesisBoxHandler.class);
+    private @Nullable IntesisBoxSocketApi intesisBoxSocketApi;
+
+    private final Map<String, String> properties = new HashMap<>();
+    private final Map<String, List<String>> limits = new HashMap<String, List<String>>();
+
+    private final IntesisDynamicStateDescriptionProvider intesisStateDescriptionProvider;
+
+    private IntesisBoxConfiguration config = new IntesisBoxConfiguration();
+
+    private double minTemp = 0.0, maxTemp = 0.0;
+
+    private boolean hasProperties = false;
+
+    private @Nullable ScheduledFuture<?> pollingTask;
+
+    public IntesisBoxHandler(Thing thing, IntesisDynamicStateDescriptionProvider intesisStateDescriptionProvider) {
+        super(thing);
+        this.intesisStateDescriptionProvider = intesisStateDescriptionProvider;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(IntesisBoxConfiguration.class);
+
+        if (!config.ipAddress.isEmpty()) {
+
+            updateStatus(ThingStatus.UNKNOWN);
+            scheduler.submit(() -> {
+                addChannel(CHANNEL_TYPE_AMBIENTTEMP, "Number:Temperature");
+                addChannel(CHANNEL_TYPE_ERRORCODE, "String");
+                addChannel(CHANNEL_TYPE_ERRORSTATUS, "String");
```

Comment:
If they are always added, can't they be specified in the XML file?

Reply:
Done